File: data-structures/typescript/src/queue.ts
import { LinkedList } from "./linked-list";

export class Queue<Type> {
  private list = new LinkedList<Type>();

  constructor(element?: Type) {
    if (element) {
      this.offer(element);
    }
  }

  size(): number {
    return this.list.getSize();
  }

  isEmpty(): boolean {
    return this.size() === 0;
  }

  peek(): Type {
    if (this.isEmpty()) {
      throw new Error("Queue empty");
    }

    return this.list.peekFirst();
  }

  poll(): Type {
    if (this.isEmpty()) {
      throw new Error("Queue empty");
    }

    return this.list.removeFirst();
  }

  offer(element: Type) {
    this.list.addLast(element);
  }
}

File: data-structures/java/src/main/java/std/data/struct/util/Queue.java
package std.data.struct.util;

import java.util.LinkedList;
import java.util.List;

public class Queue {
    private List<Object> list = new LinkedList();

    public void add(Object element) {
        this.list.add(element);
    }

    public Object poll() {
        return this.list.remove(0);
    }

    public boolean isEmpty() {
        return this.list.size() == 0;
    }

    public int size() {
        return this.list.size();
    }

    @Override
    public String toString() {
        return this.list.toString();
    }
}
